Octokit/Http/SimpleJsonSerializer.cs

```diff
@@ -88,24 +89,62 @@ public override object DeserializeObject(object value, Type type)
-                            if (attribute != null)
+                            if (_cachedEnums[type].ContainsKey(value))
                             {
-                                if (attribute.Value.Equals(value))
-                                    return field.GetValue(null);
+                                return _cachedEnums[type][value];
                             }
+                            else
+                            {
+                                //dictionary does not contain enum value and has no attribute (see below). So add it for future loops and return value
+                                // remove '-' from values coming in to be able to enum utf-8
+                                stringValue = RemoveHyphenAndUnderscore(stringValue);
+                                var parsed = Enum.Parse(type, stringValue, ignoreCase: true);
+                                _cachedEnums[type].Add(value, parsed);
+                                return parsed;
+                            }
+                        }
+                        else
+                        {
+                            //First add type to Dictionary
+                            _cachedEnums.Add(type, new Dictionary<object, object>());
+                            //then try to get all custom attributes, this happens only once per type
+                            var fields = type.GetRuntimeFields();
+                            foreach (var field in fields)
+                            {
+                                if (field.Name == "value__")
+                                    continue;
+                                var attribute = (ParameterAttribute)field.GetCustomAttribute(typeof(ParameterAttribute));
+                                if (attribute != null)
+                                {
+                                    if (attribute.Value.Equals(value))
+                                        attributeValue = field.GetValue(null);//Store value that we can return it after loop
+                                    _cachedEnums[type].Add(attribute.Value, field.GetValue(null));
+
+                                }
+                            }
+                        }
+                        //Check if we have a match for attribute
+                        if (attributeValue != null)
+                        {
+                            return attributeValue;
+                        }
+                        else
+                        {
+                            //Fallback for any reason we have no match. This sould also happens once
+                            // remove '-' from values coming in to be able to enum utf-8
+                            stringValue = RemoveHyphenAndUnderscore(stringValue);
+                            var parsed = Enum.Parse(type, stringValue, ignoreCase: true);
+                            _cachedEnums[type].Add(value, parsed);
+                            return parsed;
```

Comment:
I've ended with 
https://github.com/octokit/octokit.net/pull/1402/files#diff-2e183b0cbda11827ea1db237ac05ce00R120
Because we need to store the value that comes from the api like `blob` or `open` and not the name of the enum field `Blob`, `Open`. If we store the name of the field we every time have to parse between these two values. Please correct me if i'm wrong.